core/move.js

```diff
@@ -0,0 +1,47 @@
+/* @flow */
+
+const _ = require('lodash')
+
+/*::
+import type { Metadata } from './metadata'
+*/
+
+// Export so the following possible is possible:
+//
+//    const move = require('.../move')
+//    move(src, dst)
+//    move.child(src, dst)
+//
+module.exports = move
+move.child = child
+
+// Modify the given src/dst docs so they can be merged then moved accordingly
+// during sync.
+function move (src /*: Metadata */, dst /*: Metadata */) {
+  // moveTo is used for comparison. It's safer to take _id
+  // than path for this case, as explained in doc/developer/design.md
+  src.moveTo = dst._id
+  src._deleted = true
+
+  // Make sure newly moved docs have their fill of sync attempts
+  delete src.errors
```

Comment:
This was done in `moveFile` but not in `moveFolder`, not sure why... But I assume counting errors on a doc to be deleted doesn't make sense, so better remove it.